Modules/xASL_module_ASL.m

```diff
@@ -184,7 +184,6 @@
     elseif x.Q.NumberEchoTimes>2
         x.modules.asl.bMultiTE = true;
```

Comment:
I don't understand. The idea was to replace bMultiTE by bMultiTEQuantification right?
Because without bMultiTEQuantification, nothing should change with bMultiTE set to true?

Reply:
Not really. bMultiTE and bMultiTEQuantification are different things with different functionality, we can't replace one with another.

bMultiTEQuantification -> you wanted this now

bMultiTE -> you strongly insisted on having that 1 year ago as a subsitute for numel(uniqueEchoTime)>1 .

We can't remove bMultiTE and replace it for bMultiTEQUantification as these have a completely different purpose. I can remove bMultiTE, but it was really you who insisted on having it.

Reply:
I probably wanted bMultiTE for the reasons I now want bMultiTEQuantification?
I just give this comment for consistency.

So just to clarify, for quantification, we have Booleans if we want to process multi-PLD or multi-TE.
But there is no other reason for bMultiTE or bMultiPLD right (there was before we revamped PWI4D).

So my suggestion: we keep booleans for wanting a certain quantification to be done, which we can set to default if the user doesn't provide this, and I'm fine with defaulting to multi-TE quantification instead of processing ASL as dual-echo fMRI.

So that would mean that:

bMultiTE -> gets removed
bMultiPLD -> bMultiPLDQuantification

right? Or am I missing something (so not disagreeing with the past, just trying to accommodate to the new situation)

Reply:
bMultiPLD has this exact function. It cannot be renamed to bMUltiPLDQuantification because it has a different function.

It makes no sense to have bMultiPLDQuantification because it makes no sense to quantify multi-PLD as single PLD. Besides you would have to choose PLDs etc. So a completely different issue.

Reply:
OK, my perspective on the logics:

1. NumberEchoTimes>1 is the detection of multiple echo times, let's not use booleans for that. We have now a harmonized nPLD nEchoTimes etc definition at the start of xASL_module_ASL.
2. We used booleans bMultiTE & bMultiPLD for image processing, but now we don't need them anymore for image processing; because single-PLD & single-TE are processed equally (i.e. the automatic extension of single TE and PLD values to form the whole vector, etc). So we make our image processing ignorant of these booleans and may even remove these booleans.
3. We may need quantification booleans for multi-TE or multi-PLD (e.g., having 2 or 3 PLDs could still mean that you want to visually compare them, not quantify by fitting ATT; in fact, for QC a single-PLD quantification of the last PLD could be interesting for multi-PLD datasets).
So I suggest that we have these booleans, which we default to TRUE (quantifying multi-TE & multi-PLD using BASIL/FABBER). I would argue that `bQuantifyMultiPLD` & `bQuantifyMultiTE` would be clearer than `bMultiPLD` & `bMultiTE` which can be confusing. What do you think?

Let's move this discussion to #1543 OK?

Reply:
Mostly agree.
1. OK - I will remove bMultiTE and replace by NumberEchoTimes>1. I will do that in this issue.
2. bMultiTE is not needed anymore. bMultiPLD is needed though. THe processing is done independent of this, but at one point we have to decide to call BASIL or not. And nPLD is a bit more tricky thing than bNumberEchos. So bMultiPLD can be useful. But I can delete it if you prefer.
3. bMultiTEQUantification - we definitively need that. bMultiPLDQUantification - I would still avoid.

So on point 1 - we have an agreement and I can do that. Points 2 and 3 - do we agree or we move?

Reply:
1. OK (though check that we keep nomenclature synced for TE PLD LD)
2. OK but you are talking about quantification (doing the multi-PLD quantification using BASIL) so I would call this bQuantifyMultiPLD for clarity, and remove bMultiPLD. We default bQuantifyMultiPLD to true and if the user sets it to false, we warn that we haven't implemented this yet (we could in the future make a parameter in which the user chooses which PLDs to quantify).
3. We remove bMultiTE and keep bQuantifyMultiTE

OK?

Reply:
1. OK
2. OK. We remove bMultiPLD. I woudln't add bQuantifyMultiPLD now - cause that's a really weird option and it's not implemented anyway. And it's not even clear how this should be implemented as it really makes not much sense to quantify multiPLD as singlePLD - you don't really gain anything.
3. OK.Do we rename bMultiTEQUantification to bQuantifyMultiTE, or you just typed it wrong? I am asking because you suggested the name and it's already in the code and in the documentation - so extra work.

Reply:
1. OK
2. OK. Then in the future we have to specify how to manage quantifying multi-PLD as single-PLD (you don't seem to get it yet ;) 
clearest example is dual-PLD, 1 early and one late, one for visualizing vascular phase, one for measuring CBF. In this case, you don't want to use the first PLD when quantifying the second as multi-PLD, because this will contaminate the late PLD with vascular artifacts.
We don't do this now. Discussion for late. For now: if a user wants to quantify PLDs separately (like Mathijs did with NOMARED) the user needs to split the sequence manually. OK?
(note that my point was just having this parameter bQuantifyMultiPLD, and then saying that this is not implemented yet, and suggesting how this can be implemented, e.g. by manually splitting the sequence like Mathijs did)
3. Yes, I would call it bQuantifyMultiTE. But not very important to rename, so if you don't want extra work we keep it like it is.

Reply:
1. Done. bMultiTE replaced by x.Q.NumberEchoTimes>1. And I have fixed the dual-echo preparations - there was a bug.
2. Yes. Understand and agree - my whole point was that the user might be confused by having an option that doesn't really do anything. And when we are implementing that in the future (and I don't object here) - we might change our mind about parameters etc. But while doing this change, I have noticed that we had the option to turn of bMultiPLD - so I've updated the code and used NumberPLD>1 or bQuantifyMultiPLD where appropriate. And updated the documentation.
2b. bMultiPLD removed
2c. Fixed some small bugs in variable names that I've noticed.
3. OK. Actually, nobody has ever used this branch for doing any processing. So it's safe enough to change this now. So I did the renaming. And also in the Documentation.

All should be now fixed in the way you wanted in the latest commit.